trin-history/src/utils.rs

```diff
@@ -0,0 +1,41 @@
+use trin_core::portalnet::types::{HistoryContentKey, HeaderKey, BodyKey, ReceiptsKey};
+use trin_core::portalnet::U256;
+use sha3::{Digest, Sha3_256};
+use std::convert::TryInto;
+
+// content-id derivation function for history network
+pub fn history_network_derive_content_id(key: &String) -> U256 {
+    // decode key
+    let decoded_key = match HistoryContentKey::from_bytes(&key.as_bytes()) {
+        Ok(k) => k,
+        Err(_) => panic!("Could not decode key"),
+    };
+
+    // calculate content-id
+    match decoded_key {
+        HistoryContentKey::HeaderKey(HeaderKey { chain_id, content_type, block_hash }) => {
+            let input = U256::from(chain_id) + U256::from(content_type) + block_hash;
+            sha256(&input.to_string())
+        },
+        HistoryContentKey::BodyKey(BodyKey { chain_id, content_type, block_hash }) => {
+            let input = U256::from(chain_id) + U256::from(content_type) + block_hash;
+            sha256(&input.to_string())
+        },
+        HistoryContentKey::ReceiptsKey(ReceiptsKey { chain_id, content_type, block_hash }) => {
+            let input = U256::from(chain_id) + U256::from(content_type) + block_hash;
+            sha256(&input.to_string())
+        }
+    }
+}
+
+fn vec_to_array<T, const N: usize>(v: Vec<T>) -> [T; N] {
+    v.try_into()
+      .unwrap_or_else(|v: Vec<T>| panic!("Expected a Vec of length {} but it was {}", N, v.len()))
+}
+
+fn sha256(key: &String) -> U256 {
```

Comment:
Feels silly to redefine this in every crate. Seems like this and keccak could go into the `trin-core` utils.